### PR TITLE
Split out server-only code from api packages shared with the client

### PR DESCRIPTION
**Pull request by sttts**
Fixes staging/copy.sh.

### CODE REVIEW

**Comment by sttts** [comment]
@k8s-bot gci gke e2e test this

**Comment by sttts** [comment]
@k8s-bot cvm gke e2e test this

File: pkg/apis/rbac/validation/validation_test.go [hunk 1]
```diff
@@ -558,3 +559,10 @@ func TestValidateRoleNoResources(t *testing.T) {
 		field:   "rules[0].resources",
 	}.test(t)
 }
+
+func nonResourceURLCovers(ownerPath, subPath string) bool {
```
Comment (deads2k) [inline pkg/apis/rbac/validation/validation_test.go]: Why is this in a test file.  That seems strange.

File: plugin/pkg/auth/authorizer/rbac/rbac_test.go [hunk 2]
```diff
@@ -414,3 +421,236 @@ func (r *requestAttributeBuilder) URL(url string) *requestAttributeBuilder {
 func (r *requestAttributeBuilder) New() authorizer.AttributesRecord {
 	return r.request
 }
+
+type AuthorizationRuleResolver interface {
```
Comment (deads2k) [inline plugin/pkg/auth/authorizer/rbac/rbac_test.go]: duplication for some reason?

**Comment by deads2k** [comment]
I'm fine with the moves.  I'm less fine with the duplication.  Why do we have the escalation logic duplicated?

File: pkg/apis/rbac/validation/validation_test.go (hunk 1 above)
Comment (sttts) [inline pkg/apis/rbac/validation/validation_test.go]: It's only used here.

File: plugin/pkg/auth/authorizer/rbac/rbac_test.go (hunk 2 above)
Comment (sttts) [inline plugin/pkg/auth/authorizer/rbac/rbac_test.go]: this should not be here. Good spot.

Comment (sttts) [inline plugin/pkg/auth/authorizer/rbac/rbac_test.go]: fixed

File: pkg/apis/rbac/validation/validation_test.go (hunk 1 above)
Comment (sttts) [inline pkg/apis/rbac/validation/validation_test.go]: We even test it above. Should we throw it away?

Comment (sttts) [inline pkg/apis/rbac/validation/validation_test.go]: in fact we have a copy in the registry. Will move the test and remove this instance.

Comment (sttts) [inline pkg/apis/rbac/validation/validation_test.go]: done

**Comment by deads2k** [comment]
lgtm

**Comment by sttts** [comment]
Rebased after apimachinery PR merged.

**Comment by sttts** [comment]
@k8s-bot bazel test this

**Pull request merged**